src/yaml-helpers.h

```diff
@@ -41,7 +42,9 @@
 }
 #define YAML_SCALAR_PLAIN(event_ptr, emitter_ptr, scalar) \
 { \
-    yaml_scalar_event_initialize(event_ptr, NULL, (yaml_char_t *)YAML_STR_TAG, (yaml_char_t *)scalar, strlen(scalar), 1, 0, YAML_PLAIN_SCALAR_STYLE); \
+    size_t _length = strlen(scalar); \
+    g_assert(_length < G_MAXINT); \
+    yaml_scalar_event_initialize(event_ptr, NULL, (yaml_char_t *)YAML_STR_TAG, (yaml_char_t *)scalar, (int)_length, 1, 0, YAML_PLAIN_SCALAR_STYLE); \
```

Comment:
nitpick (non-blocking): I guess we could also just pass `-1` as the `length` parameter and letting `libyaml` deal with it itself, avoiding any extra calls. But it's fine as-is, too.

See: https://github.com/yaml/libyaml/pull/102